apptests/kustomize/kustomize.go

```diff
@@ -0,0 +1,86 @@
+package kustomize
+
+import (
+	"os"
+
+	"github.com/drone/envsubst"
+	"sigs.k8s.io/kustomize/api/krusty"
+	"sigs.k8s.io/kustomize/api/resmap"
+	"sigs.k8s.io/kustomize/api/resource"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+)
+
+type Kustomize struct {
+	directory   string            // The directory where the kustomization file is located
+	substitutes map[string]string // The map of environment variables to substitute in the resources
+	resources   resmap.ResMap     // The map of resources generated by the kustomization process
+}
+
+// New creates a new Kustomize instance with the given directory and substitutes.
+func New(dir string, subs map[string]string) *Kustomize {
+	return &Kustomize{
+		directory:   dir,
+		substitutes: subs,
+		resources:   resmap.New(),
+	}
+}
+
+// Build runs the kustomization process on the directory and substitutes the
+// environment variables in the resources.
+func (k *Kustomize) Build() error {
+	for k, v := range k.substitutes {
+		os.Setenv(k, v)
+	}
```

Comment:
This function has a side effect of modifying the process environment where it runs, potentially being insecure as it doesn't revert the changes. I think we should add a `defer` call that will revert any possible changes.

Another thing is that this function cannot be called in parallel and so I think it should be lock protected.

Reply:
Use `envsubst.Eval` instead - do not update the environment like this.